core/src/main/java/org/apache/accumulo/core/iterators/WrappingIterator.java

```diff
@@ -58,21 +58,21 @@ protected void setSource(SortedKeyValueIterator<Key,Value> source) {
   public Key getTopKey() {
     if (seenSeek == false)
       throw new IllegalStateException("never been seeked");
-    return getSource().getTopKey();
+    return source.getTopKey();
```

Comment:
This could possibly break iterators that extend WrappingIterator and override getSource().  I don't think we should make this change.

Maybe it would be better to deprecate WrappingIterator and introduce another version for users that more efficient.  I don't think we have a lot of wiggle room to make this one more efficient without possibly breaking user iterators.